website/src/components/Toc.vue

```diff
@@ -26,6 +26,38 @@ export default {
     filteredHeadings() {
       return this.headings.filter(heading => heading.level < 4)
     }
+  },
+
+  data() {
+    return {
+      currentHash: null
+    }
+  },
+
+  methods: {
+    handleScrolling() {
+      this.hTags.forEach(el => {
+        const top = window.pageYOffset
+        const distance = top - el.offsetTop
+        const hash = el.getElementsByTagName('a')[0].hash
+        if (distance < 50 && distance > -50 && this.currentHash != hash) {
```

Comment:
Personally, I would recommend using [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) to handle these, it looks like:

```js
data() {
  return {
    currentHash: null,
    observer: null
  }
}

mounted() {
  this.observer = new IntersectionObserver(([firstEntry]) => {
    if (firstEntry.isIntersecting) {
      const hash = `#${firstEntry.id}`
      history.pushState(null, null, hash)
      this.currentHash = hash
    }
  })

  const hLevels = [2, 3]
  hLevels.forEach(hLevel => document.querySelectorAll(`h${hLevel}`).forEach(el => observer.observe(el)))
}

beforeDestroy() {
  this.observer.disconnect()
}
```

Reply:
doing it in this way will be better but even so it will still cause some problems
```
mounted() {
  	this.observer = new IntersectionObserver((entries) => {
      let hash = entries.length === 1 ? `#${entries[0].target.id}` : null
      history.pushState(null, null, hash)
      this.currentHash = hash
    }, {rootMargin: `0px 0px ${document.body.offsetHeight-50}px 0px`})

    const hLevels = [2, 3]
    hLevels.forEach(hLevel => document.querySelectorAll(`h${hLevel}`).forEach(el => this.observer.observe(el)))
  },
```
thanks for the info

Reply:
Sorry, the initial example is wrong, check out the latest example 😃 
Still there's a little difference between scroll listener and IntersectionObserver but that's fine (in my opinion).

Reply:
No I think that IntersectionObserver is way more performant than scroll listener. Maybe I will change to IntersectionObserver

Reply:
Maybe you can just use this little library: https://github.com/camwiegert/in-view

Reply:
I just tried using this library but it's not good because of https://github.com/camwiegert/in-view#performance 
> throttled (maximum once every 100ms)

If you don't scroll slowly it wouldn't work nicely.

I think that how I have done it is the best for now as for performance it's not a problem because the website doesn't contain anything heavy (And it works good for me while using a laptop that was made before 2006)

Reply:
Oh I thought it was using IntersectionObserver already 😅 so I was wrong.

Reply:
I was able to implement it with IntersectionObserver I just need to perfect it